sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClientTest.java

```diff
@@ -964,6 +966,11 @@ public RntbdRequestRecord request(final RntbdRequestArgs requestArgs) {
             return requestRecord;
         }
 
+        @Override
+        public OpenConnectionRntbdRequestRecord openConnection(Uri addressUri) {
```

Comment:
Is it possible to add a emulator test that uses the new API. Then do 10 random create items and inspect the diagnostics of all the operations and verify none had to create the connection?

Reply:
that is a good suggestion, I have only done manual testing, will add a test as well

Reply:
added tests to verify for document requests happened after, no new connection is being created